### PR TITLE
1.4.1 import single scene

### LINKED ISSUE #8: Additional changes
- [ ] Investigate Icon migration
- [ ] Single import of scene is not linking tokens or notes properly.
- [ ] Should be some sort of progress when creating documents
- [ ] v9 is not linking Actor links in Journal Entries.

### LINKED ISSUE #16: Add support for importing actors and notes when importing individual scenes

### LINKED ISSUE #17: Search icon doesn't highlight when searching

### LINKED ISSUE #19: v11 will occasionally add a double icon in the manage modules list
Not sure what's going on, but occasionally when you open the Manage Modules config, it will double-up on the Forge compendium icon.

### LINKED ISSUE #21: [Feature Request] Scene configuration options when importing scenes
Feature requested [here](https://forums.forge-vtt.com/t/setting-defaults-loading-content-added-numerous-scenes-to-navigation-bar/90429)

It may be possible to add additional configuration options before the first import that could apply to each scene in the import, to help streamline certain workflows, e.g.
- [ ] Activate/deactivate 'show in navigation' for scenes before import.

### LINKED ISSUE #23: Compendium Browser on v10 does not populate individual item/actors/scene pages with respective info
### Steps: 
- create a world on v10
- install relevant modules
- open compendium library
- Navigate to an actor/scene/item etc and click on an individual item

➡️ Result: Nothing displays,
Note: Importing also doesn't work 

Screen Shot 2023-08-30 at 1 56 16 PM

### LINKED ISSUE #24: Investigate Compendium Library module on v9 is not prompting to install
- create a v9 world
- launch game, navigate to manage modules
- filter for `dnd`
- select one

➡️ Pop up will ask to install dependencies but is missing the compendium browser one
See second image on v10 showing dependencies required including compendium browser

Screen Shot 2023-08-30 at 2 25 06 PM
Screen Shot 2023-08-30 at 2 36 11 PM

### LINKED ISSUE #25: Lock icon is hard to see on compendiums with lighter backgrounds
- open compendium library 
- right click on compendium with a lighter background
- choose Configure Permissions
- Change All Players > Not Allowed & Save

➡️ Lock icon is challenging to see on a lighter background > See `Lost Mine` in image compared to `Monstrous` 

Screen Shot 2023-08-31 at 2 27 05 PM

### LINKED ISSUE #26: sub directory class is getting interference from other modules
image

### LINKED ISSUE #28: Compendium Library button is not obvious to user to go back to main page
Possible suggestions: 
- change terminology to `Back to Compendium Browser`
- Include an obvious BACK icon ⏪ or ⬅️
- modify color?

Screen Shot 2023-09-06 at 3 01 39 PM

### LINKED ISSUE #29: Clicking on Actor/Item name in top bar does not go anywhere
- Launch compendium library in Foundry
- Open an actors or item list
- choose any actor or item
- click on individual actor/item name in top bar 

➡️ Nothing happens; the green navigation indicator on the left hand side just moves to the higher level in hierarchy

https://github.com/ForgeVTT/forge-compendium-browser/assets/5169836/3ae18d68-51e1-4099-b6bf-8e1c7801db5c

### LINKED ISSUE #30: Clicking on letter of alphabet should auto scroll actors/items list to that letter
- open compendium library
- click on Actors
- on left hand side navigation, click on any letter of the alphabet

➡️ Result: nothing happens on right hand side navigation, left hand navigation expands the actors/items under that letter
*️⃣ Expected: clicking on a letter on the left hand side should auto scroll the actors page to that letter

### LINKED ISSUE #31: Scenes are not auto appearing on top bar of Foundry game after importing
- open compendium browser
- choose any scene & import
- navigate to scenes & confirm scene has been imported
➡️ Result: scene is not displayed across top of game
Workaround: refresh game and scene will appear

https://github.com/ForgeVTT/forge-compendium-browser/assets/5169836/e87ec5cd-ac20-4ec9-a441-2c28ef958936

### LINKED ISSUE #32: Match terminology of Permissions vs Ownership in Compendium library
Update strings: 
- right click on compendium book cover
- main page on compendium 
- modal pop up after clicking on lock icon

Screen Shot 2023-09-06 at 3 22 09 PM
Screen Shot 2023-09-06 at 3 21 58 PM
Screen Shot 2023-09-06 at 3 58 17 PM

### LINKED ISSUE #34: Rebulding book structure causes actors/items not to load and freezes
- open compendium library on v11 w/ version 1.4.0
- right click on any compendium
- choose rebuild
- attempt to click on actors and select any to open character sheet

➡️ Result: Compendium library freezes, character sheet doesn't load, console errors

Screen Shot 2023-09-07 at 11 07 12 AM
Screen Shot 2023-09-07 at 11 12 15 AM

### LINKED ISSUE #35: Search button doesn't actually seem to do anything functional
- open compendium browser and choose any compendium
- click on search button
- type in any search term ie `dragon` 
- click on search button

➡️ clicking on search button doesn't seem to functionally do anything, it doesn't trigger the search as the search results automatically adapts to what's input in the search field 

https://github.com/ForgeVTT/forge-compendium-browser/assets/5169836/480dbf6a-679e-4e5c-9a75-c40f33dd963a

### LINKED ISSUE #36: Compendium name is missing from notification that it's done building
➡️ currently displaying `{name}`
- can trigger by creating a new game and activating your compendium modules
- or by rebuilding compendium 

Screen Shot 2023-09-09 at 9 58 32 AM

### LINKED ISSUE #37: Incompatible warning on version in Module Management settings
- loaded up latest version on v11 and got this warning

Screen Shot 2023-09-09 at 10 01 55 AM

### PR DESCRIPTION
**Pull request by ironmonk108**
closes #37 
closes #36 
closes #35 
closes #34 
closes #32
closes #31
closes #30 
closes #29 
closes #28 
closes #26 
closes #25 
closes #24 
closes #23 
closes #21 
closes #19 
closes #17 
closes #16 
closes #8

### CODE REVIEW

File: lang/en.json
Comment (rikmarais) [inline lang/en.json:1]: d90fb52020586500b72d3d016475c0aa10ca8835 changes this file's line endings from LF to CRLF, but I'm not sure if it's intentional

File: module.json [hunk 1]
```diff
@@ -1,8 +1,9 @@
 {
   "name": "forge-compendium-browser",
+  "id": "forge-compendium-browser",
   "title": "The Forge: Compendium Library",
   "description": "Browse your compendium books like they were meant to be viewed. The Forge's Compendium Library lets you navigate your D&amp;D Beyond imported books in a beautiful and easy to use interface!",
-  "version": "1.4.1",
+  "version": "1.4.0",
```
Comment (rikmarais) [inline module.json:6]: Do we want to roll back the version? It shouldn't be a problem if we do, just want to make sure this was intentional. Adding `id` is good 👍

File: src/apps/compendium-browser-app.js
Comment (rikmarais) [inline src/apps/compendium-browser-app.js:1]: fc664bb4cdcf4d608f2a6353dfe02efbd5e80313 changes this file's line endings from LF to CRLF, and is likely unintentional

File: lang/en.json [hunk 2]
```diff
@@ -1,32 +1,55 @@
...
-    "ForgeCompendiumBrowser.Wall": "Walls"
-}
+{
+    "ForgeCompendiumBrowser.same-name.name": "Auto open with same name",
+    "ForgeCompendiumBrowser.same-name.hint": "Automatically open documents with the same name as the listing name",
+    "ForgeCompendiumBrowser.set-navigate.name": "Set Scene Navigate",
+    "ForgeCompendiumBrowser.set-navigate.hint": "Set the scene as navigatable when the scene is imported",
+    "ForgeCompendiumBrowser.OpenCompendiumBrowser": "Open Compendium Library",
+    "ForgeCompendiumBrowser.ForgeCompendiumLibrary": "Forge Compendium Library",
+    "ForgeCompendiumBrowser.LibraryMessage": "This is your library of compendium books.  If you have no books listed, please use the {link} from the Bazaar to import your books and enable the module in your world",
```
Comment (rikmarais) [inline lang/en.json:8]: Double space before `.  If`

File: src/hierarchy.js
Comment (rikmarais) [inline src/hierarchy.js:1]: 693f622f158ee389062a5c654bbcd08f6e779afd changes this file's line endings from LF to CRLF, but it looks unintentional

File: src/main.js
Comment (rikmarais) [inline src/main.js:1]: d17d3a2972b47ede7429efd5444aed31cfc5d783 changes this file's line endings from LF to CRLF, but I don't think that it is intentional

File: src/settings.js
Comment (rikmarais) [inline src/settings.js:1]: 469723738b93927b1174c2839fd474971e5ebda9 changes this file's line endings from LF to CRLF, but it looks unintentional

**Review by rikmarais — CHANGES_REQUESTED** [review]
I think that there were some accidental line ending changes in this PR which will need to be changed back.

Ignoring whitespace changes, I'm going to say tentative approval. The code looks good, but I notice that the PR is still in draft state, so I didn't heavily test it yet.

@ironmonk88 I can switch the line endings where necessary if you're alright with that, but are there any pending changes or can I move this one out of draft status and give it a solid bash?

File: module.json (hunk 1 above)
Comment (ironmonk88) [reply module.json:6]: Yep.  KaKaRoTo said there's no reason to go with 1.4.1, it should be 1.4.0

File: lang/en.json (hunk 2 above)
Comment (ironmonk88) [reply lang/en.json:8]: Because double spaces are awesome!

File: src/importBook.js [hunk 3]
```diff
@@ -299,15 +300,21 @@ export class ImportBook {
                             if (isV10) {
                                 token.actorId = actor._id;
                                 if (
-                                    (!token.texture?.src || token.texture?.src === "icons/svg/mystery-man.svg") &&
+                                    (!token.texture?.src ||
+                                        token.texture?.src === "icons/svg/mystery-man.svg" ||
+                                        token.texture?.src?.includes("placeholder") ||
+                                        token.texture?.src?.includes("-token")) &&
```
Comment (rikmarais) [inline src/importBook.js:306]: I'm not certain about the `.includes` here as opposed to `.endsWith`, but as long as the data adheres to this convention, it should be fine.

**Review by rikmarais — APPROVED** [review]

**Comment by rikmarais** [comment]
Looks good @ironmonk88! Thanks for the changes, and nice work 💪 
I did have one slight concern surrounding the common conventions for defining token data, thinking that it might be too much of a generalization to replace any token that contains `-token` in the src with the actor image, but this is very dependent on the data available to the compendium browser. 

From the ddb data, it should be reliable, but the browser could be used for other content as well. If that feels like it could be a problem, we might want to introduce a setting in the future that turns this auto-replace on or off.

**Comment by ironmonk88** [comment]
Good call on the token.
I added a ticket so we don't forget.
For now it should be fine though since only ddb books are available.  But as soon as it becomes a concern, we should deal with it.

**Pull request merged**